docs/topics/whatsnew-eap.md

```diff
@@ -572,7 +572,55 @@ We would appreciate any feedback you may have!
 * [Enable the Send usage statistics option](https://www.jetbrains.com/help/idea/settings-usage-statistics.html) to
   allow JetBrains to collect anonymous data about K2 usage.
 
-## Kotlin/Native: resolving conflicts with Objective-C methods
+## Kotlin/JVM
+
+This version brings the following changes:
+
+* [Generate lambda functions using invokedynamic](#generate-lambda-functions-using-invokedynamic)
+* [The kotlinx-metadata-jvm library is now Stable](#the-kotlinx-metadata-jvm-library-is-now-stable)
+
+### Generate lambda functions using invokedynamic
+
+Kotlin %kotlinEapVersion% introduces a new default method for generating lambda functions using `invokedynamic`.
+This change reduces the binary sizes of applications compared to the traditional anonymous class generation.
+
+Since the first version, Kotlin has generated lambdas as anonymous classes. However, starting from [Kotlin 1.5](whatsnew15.md#lambdas-via-invokedynamic),
+the option for `invokedynamic` generation was available by using the `-Xlambdas=indy` compiler flag. In Kotlin %kotlinEapVersion%,
+`invokedynamic` has become the default method for lambda generation. This method produces lighter binaries and aligns
+Kotlin with JVM optimizations, ensuring applications benefit from ongoing and future improvements in JVM performance.
+
+Currently, it has three limitations compared to ordinary lambda compilation:
+
+* A lambda compiled into `invokedynamic` is not serializable.
+* Calling `toString()` on such a lambda produces a less readable string representation.
```

Comment:
Sure, we can put it as the last point, but I don't think it can be clarified in a simple way without including a long example which would make it far less readable. 

I think we should include it after it leaves RC in the docs, but in the what's new section, it would feel a bit too much to me.